web/app/components/product/avatar.gts

```diff
@@ -0,0 +1,59 @@
+import Component from "@glimmer/component";
+import getProductID from "hermes/utils/get-product-id";
+import { inject as service } from "@ember/service";
+import FlightIcon from "@hashicorp/ember-flight-icons/components/flight-icon";
+import ProductAreasService from "hermes/services/product-areas";
+import { assert } from "@ember/debug";
+import { HermesAvatarSize } from "hermes/types/avatar-size";
+
+interface ProductAvatarComponentSignature {
+  Element: HTMLDivElement;
+  Args: {
+    product: string;
+    size?: `${HermesAvatarSize}`;
+  };
+  Blocks: {
+    default: [];
+  };
+}
+
+export default class ProductAvatarComponent extends Component<ProductAvatarComponentSignature> {
+  @service declare productAreas: ProductAreasService;
+
+  private get productID(): string {
+    const productID = getProductID(this.args.product);
+    assert("productID must edxist", productID);
+    return productID;
+  }
+
+  private get sizeIsMedium() {
+    return this.args.size === HermesAvatarSize.Medium;
+  }
+
+  private get sizeIsLarge() {
+    return this.args.size === HermesAvatarSize.Large;
+  }
```

Comment:
Curious about this pattern vs. checking the value in the template using a more generic getter that just returns the size (like was added in the thumbnail component in this PR) - seems like that could be more scalable?

Reply:
Most of it was the convenience of having everything in one file in the early phase. But it's worth noting that Tailwind's [dynamic class rules](https://tailwindcss.com/docs/content-configuration#dynamic-class-names) limit our efficiency a bit.* Still, we're past the early phase, so we can be smarter here. I refactored to use CSS classes.

* Of course, there are workarounds. I think safe-listing the `w-` and `h-` classes would allow us to do stuff like `class="h-{{this.size}}"` if that were ever preferable to CSS.

Reply:
Cool, sounds good 👍